DebtNote/Controllers/UserController.cs

```diff
@@ -26,39 +26,40 @@ public UserController(ApplicationContext context, IUserService userService)
         [HttpGet("id")]
         [ProducesResponseType(typeof(User), StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status404NotFound)]
-        public JsonResult GetById(int id)
+        public IActionResult GetById(int id)
         {
-            return new JsonResult(UserService.GetById(id));//do smth for 404
+            var result = UserService.GetById(id);
+            return result == null ? NotFound() : Ok(result);
         }
 
         [HttpPost]
         [ProducesResponseType(StatusCodes.Status201Created)]
-        public JsonResult CreateUser(User user) => new JsonResult(UserService.CreateUser(user));
+        public IActionResult CreateUser(User user)
+        {
+            UserService.CreateUser(user);
+            return CreatedAtAction(nameof(GetById),new {id=user.Id},user);
+        } 
         
         [HttpPut("{id}")]
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
-        public JsonResult UpdateUser(User user) => new JsonResult(UserService.UpdateUser(user)); //TODO: fix error 500
-        //public async Task<IActionResult> UpdateUser(int id, User user)
-        //{
-        //    if(id!= user.Id) return BadRequest();
-        //    _context.Entry(user).State = EntityState.Modified;
-        //    await _context.SaveChangesAsync();
-        //    return NoContent();
-        //}
+        public IActionResult UpdateUser(int id, User user)
+        {
+            if (id != user.Id) return BadRequest();
+            UserService.UpdateUser(user);
```

Comment:
no point in private properties, just create `_userService` private variable. This style is confusing, nobody uses private properties

Reply:
Is that about user UserService right?